admin/pre_reg/views.py

```diff
@@ -102,16 +103,22 @@ class DraftDetailView(PreregAdmin, DetailView):
     context_object_name = 'draft'
 
     def get_object(self, queryset=None):
+        draft = DraftRegistration.load(self.kwargs.get('draft_pk'))
+        self.checkout_files(draft)
         try:
-            return serializers.serialize_draft_registration(
-                DraftRegistration.load(self.kwargs.get('draft_pk'))
-            )
+            return serializers.serialize_draft_registration(draft)
         except AttributeError:
             raise Http404('{} with id "{}" not found.'.format(
                 self.context_object_name.title(),
                 self.kwargs.get('draft_pk')
             ))
 
+    def checkout_files(self, draft):
```

Comment:
So I think this is what you'd want to commit and begin a new transaction in if the problem ends up being on Production. I imagine the checkin view as well.